test/lib/test_fifo.py

```diff
@@ -62,3 +68,57 @@ async def target(sim: TestbenchContext):
         with self.run_simulation(fifoc) as sim:
             sim.add_testbench(source)
             sim.add_testbench(target)
+
+
+class TestWideFifo(TestCaseWithSimulator):
+    async def source(self, sim: TestbenchContext):
+        cycles = 100
+
+        for _ in range(cycles):
+            await self.random_wait_geom(sim, 0.5)
+            count = random.randint(1, self.write_width)
+            data = [const_of(random.randrange(2**self.bits), self.shape) for _ in range(self.write_width)]
+            await self.circ.write.call(sim, count=count, data=data)
+            await sim.delay(2e-9)
```

Comment:
Usage of `2e-9` is not very clear to me. Why that value and not `1e-9`? Shouldn't we use some variable with cycle length instead?

Reply:
Because 1e-9 is used in a different testbench process. This is used to enforce execution order without synchronization primitives, which sadly are not provided (?) by Amaranth.